submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,352 @@
+      this.selectSortMethod(target);
+      this.renderDataWithFilters.usersDataForRender = this.usersDataForFilter;
+      this.renderDataWithFilters.renderProfiles();
+    });
+  }
+
+  sortByGender(gender) {
+    this.usersDataForFilter = [...this.usersByDefault];
+    const selectGender = {
+      allGenders: () => {
+        this.usersDataForFilter = this.sortedUsers = [...this.usersByDefault];
+      },
+      male: () => {
+        this.usersDataForFilter = this.usersDataForFilter.filter(
+          user => user.gender == this.sortByGenderTypes.male
+        );
+        this.sortedUsers = [...this.usersDataForFilter];
+      },
+      female: () => {
+        this.usersDataForFilter = this.usersDataForFilter.filter(
+          user => user.gender == this.sortByGenderTypes.female
+        );
+        this.sortedUsers = [...this.usersDataForFilter];
+      },
+    };
+
+    return selectGender[gender]();
+  }
+
+  sortByContent(sortType, sortTypes, sortBy) {
+    if (this.sortedUsers.length !== 0) {
+      this.usersDataForFilter = [...this.sortedUsers];
+    } else {
+      this.sortedUsers = [...this.usersByDefault];
+    }
+
+    const typesOfSort = {
+      [sortTypes.byDefault]: () => {
+        this.usersDataForFilter = [...this.sortedUsers];
+      },
+      [sortTypes.inAscending]: () => {
+        if (sortBy == this.sortIndicators.name)
+          this.usersDataForFilter.sort((a, b) =>
+            a.name.first.localeCompare(b.name.first)
+          );
+        if (sortBy == this.sortIndicators.age)
+          this.usersDataForFilter.sort((a, b) =>
+            a.dob.age.toString().localeCompare(b.dob.age.toString())
+          );
+      },
+      [sortTypes.inDescending]: () => {
+        if (sortBy == this.sortIndicators.name)
+          this.usersDataForFilter.sort((a, b) =>
+            b.name.first.localeCompare(a.name.first)
+          );
+        if (sortBy == this.sortIndicators.age)
+          this.usersDataForFilter.sort((a, b) =>
+            b.dob.age.toString().localeCompare(a.dob.age.toString())
+          );
+      },
+    };
+    return typesOfSort[sortType]();
+  }
+}
+
+class Users {
+  constructor(renderer, filter) {
+    this.renderData = renderer;
+    this.filterData = filter;
+    this.apiURL = `https://randomuser.me/api/`;
+    this.errorMessage = `Uh oh, something has gone wrong. Please tweet us @randomapi about the issue. Thank you.`;
+  }
+
+  async getUsersData() {
+    try {
+      const response = await fetch(this.apiURL + `?results=100&seed=abc`);
+      if (!response.ok) throw new Error(errorMessage);
+      const usersData = await response.json();
+
+      this.renderData.usersDataForRender = this.filterData.usersDataForFilter =
+        usersData.results;
+
+      this.renderData.renderProfiles();
+      this.filterData.startFilter();
+    } catch (err) {
+      console.error(err);
+    }
+  }
+}
+
+function animation() {
+  const animationDuration = 7000;
```

Comment:
As far as I remember this is duration of intro? If so, change name please

Reply:
ok, renamed to ```animationDurationOfIntro```